file/nifile.yml

```diff
@@ -0,0 +1,816 @@
+      version:
+        type: integer
+        description: The version of the available operation
+    required: [available]
+    example:
+      available: true
+      version: 1
+  V1Operations:
+    description: V1 operations
+    type: object
+    properties:
+      operations:
+        description: >-
+          Available operations in the v1 version of the API:
+
+          - deleteFiles: The ability to delete uploaded files
+
+          - downloadData: The ability to download file data
+
+          - listFiles: The ability to query and list available files and service groups
+
+          - updateMetadata: The ability to update file metadata properties
+
+          - uploadFiles: The ability to upload files
+        type: object
+        properties:
+          deleteFiles:
+            $ref: '#/definitions/Operation'
+          downloadData:
+            $ref: '#/definitions/Operation'
+          listFiles:
+            $ref: '#/definitions/Operation'
+          updateMetadata:
+            $ref: '#/definitions/Operation'
+          uploadFiles:
+            $ref: '#/definitions/Operation'
+
+parameters:
+  id:
+    in: path
+    name: id
+    description: The ID of the file
+    type: string
+    required: true
+    x-example: '4afb2ce3741fe11d88838cc9'
+
+responses:
+  Query:
+    description: OK
+    schema:
+      description: The result of a file query
+      title: QueryResponse
+      type: object
+      properties:
+        _links:
+          description: >-
+            The links that apply to the collection of files for a service group:
+
+            - deleteFiles: Link to delete multiple files from the service group using a POST
+
+            - query: Link to query for available files in the service group using a POST
+
+            - search: Link to retrieve a filtered list of files in the service group using a GET
+
+            - self: Link to the current service group
+
+            - upload: Link to upload files to the service group using a POST
+          type: object
+          required: [self]
+          additionalProperties:
+            $ref: '#/definitions/Link'
+          properties:
+            self:
+              $ref: '#/definitions/Link'
+          example:
+            deleteFiles:
+              href: /nifile/v1/service-groups/Default/delete-files
+            query:
+              href: /nifile/v1/service-groups/Default/query-files
+            search:
+              href: /nifile/v1/service-groups/Default/files
+            self:
+              href: /nifile/v1/service-groups/Default/files
+            upload:
+              href: /nifile/v1/service-groups/Default/upload-files
+        availableFiles:
+          description: The list of files returned by the query
+          type: array
+          items:
+            $ref: '#/definitions/FileMetadata'
+        totalCount:
+          description: The total number of files that match the query regardless of skip and take
```

Comment:
I think I asked this in one of Phil's reviews, but I haven't seen if he replied to it: is there a way to make it clear that skip and take are parameters? If not, maybe add "values" to the end of the sentence.